src/coreclr/inc/clrconfigvalues.h

```diff
@@ -765,6 +765,11 @@ RETAIL_CONFIG_DWORD_INFO(EXTERNAL_EnableSSE41,        W("EnableSSE41"),        1
 RETAIL_CONFIG_DWORD_INFO(EXTERNAL_EnableSSE42,        W("EnableSSE42"),        1, "Allows SSE4.2+ hardware intrinsics to be disabled")
 RETAIL_CONFIG_DWORD_INFO(EXTERNAL_EnableSSSE3,        W("EnableSSSE3"),        1, "Allows SSSE3+ hardware intrinsics to be disabled")
 RETAIL_CONFIG_DWORD_INFO(EXTERNAL_EnableX86Serialize, W("EnableX86Serialize"), 1, "Allows X86Serialize+ hardware intrinsics to be disabled")
+RETAIL_CONFIG_DWORD_INFO(EXTERNAL_EnableAVX512F,      W("EnableAVX512F"),      1, "Allows AVX512F+ hardware intrinsics to be disabled")
+RETAIL_CONFIG_DWORD_INFO(EXTERNAL_EnableAVX512CD,      W("EnableAVX512CD"),      1, "Allows AVX512CD+ hardware intrinsics to be disabled")
+RETAIL_CONFIG_DWORD_INFO(EXTERNAL_EnableAVX512BW,      W("EnableAVX512BW"),      1, "Allows AVX512BW+ hardware intrinsics to be disabled")
+RETAIL_CONFIG_DWORD_INFO(EXTERNAL_EnableAVX512DQ,      W("EnableAVX512DQ"),      1, "Allows AVX512DQ+ hardware intrinsics to be disabled")
+RETAIL_CONFIG_DWORD_INFO(EXTERNAL_EnableAVX512VL,      W("EnableAVX512VL"),      1, "Allows AVX512VL+ hardware intrinsics to be disabled")
```

Comment:
Due to the way `InstructionSet_*` checks work today, I think AVX512-VL might need "special handling".

In particular, each intrinsic maps to a given `InstructionSet_*` and that `InstructionSet_*` gets a bit in the crossgen metadata indicating whether a method requires or opportunistically uses a given ISA.

Since AVX512-VL is "special" in that it is its own ISA extending AVX512F, but also any other supported `AVX512-*` ISA, we need to either:
A. Explicitly declare those other ISAs similarly to what we do for X64. That is, `InstructionSet_AVX512CD_VL`, `InstructionSet_AVX512BW_VL`, etc
B. Come up with a special way to represent and check for them (such as tracking the intrinsic as `InstructionSet_AVX512CD` and having a `HW_Flag_AVX512VL` flag)

The former is consistent with how we already handle things for `X64` and somewhat simplifies the way intrinsics are checked for. It also more closely matches how the ISAs will have to be exposed from managed.

However, the latter takes up less overall "space" to encode the information around and better matches how hardware is required to check for support. Theoretically if we went with this approach we could also simplify how `InstructionSet_*_X64` was encoded/worked.

It's likely worth getting input from @davidwrighton on if one approach or the other is better.

Reply:
I've made some tentative changes here while we wait for @davidwrighton . I chose option1 from above for now. Let me know if this is what you had in mind

Reply:
Thanks!

We'll want to consider the implications of each approach in relation to the `DOTNET_Enable*` environment variables and with how it impacts the ISAs users can pass into NAOT.

Hopefully we can come up with something that works well for all the scenarios here.

Reply:
Also, do you have a preference for(i.e., if the 2nd looks cleaner)

```
if ((cpuidInfo[EBX] & (1 << 16)) != 0)                     // AVX512F
{
     CPUCompileFlags.Set(InstructionSet_AVX512F);

      if ((cpuidInfo[EBX] & (1 << 31)) != 0)                 // AVX512VL
      {
               CPUCompileFlags.Set(InstructionSet_AVX512F_VL);

                if ((cpuidInfo[EBX] & (1 << 30)) != 0)             // AVX512BW
                {
                        CPUCompileFlags.Set(InstructionSet_AVX512BW_VL);
                 }

                 if ((cpuidInfo[EBX] & (1 << 28)) != 0)             // AVX512CD
                 {
                         CPUCompileFlags.Set(InstructionSet_AVX512CD_VL);
                 }

                 ..............................

      }
       if ((cpuidInfo[EBX] & (1 << 30)) != 0)                 // AVX512BW
     {
            CPUCompileFlags.Set(InstructionSet_AVX512BW);
     }

     if ((cpuidInfo[EBX] & (1 << 28)) != 0)                 // AVX512CD
     {
             CPUCompileFlags.Set(InstructionSet_AVX512CD);
     }

      .............
}
```

vs


```
if ((cpuidInfo[EBX] & (1 << 16)) != 0)                     // AVX512F
 {
           CPUCompileFlags.Set(InstructionSet_AVX512F);
            bool isAVX512_VLSupported = false;

            if ((cpuidInfo[EBX] & (1 << 31)) != 0)                 // AVX512VL
            {
                   CPUCompileFlags.Set(InstructionSet_AVX512F_VL);
                   isAVX512_VLSupported = true;
             }

             if ((cpuidInfo[EBX] & (1 << 30)) != 0)                 // AVX512BW
             {
                   CPUCompileFlags.Set(InstructionSet_AVX512BW);
                    if (isAVX512_VLSupported)
                     {
                            CPUCompileFlags.Set(InstructionSet_AVX512BW_VL);// AVX512BW_VL
                      }
             }

             if ((cpuidInfo[EBX] & (1 << 28)) != 0)                 // AVX512CD
             {
                     CPUCompileFlags.Set(InstructionSet_AVX512CD);
                     if (isAVX512_VLSupported)
                     {
                             CPUCompileFlags.Set(InstructionSet_AVX512CD_VL);// AVX512CD_VL
                     }
             }
             ...........................
}
```

Reply:
I don't have a preference here. I'd guess the second is slightly less error prone since it isn't dependent on rechecking specific bits.

Reply:
@tannergooding Are we still waiting to hear back on this or do I make other changes assuming AVX512*_VL design?

Reply:
I think we can assume the `AVX512*_VL` design for now and can fix up later if we need to.

Reply:
Yes, I think the current approach will work, and if we need to change course in the future, we can address the problem then.